neqo-transport/src/connection.rs

```diff
@@ -1157,6 +1162,24 @@ impl Connection {
         Ok(())
     }
 
+    fn handle_max_data(&mut self, maximum_data: u64) {
+        let conn_was_blocked = self.flow_mgr.borrow().conn_credit_avail() == 0;
```

Comment:
Now that I understand this better, maybe you can short-circuit the `conn_was_blocked` test.

Reply:
eh, gonna leave it. clarity more important than every last cycle here I'd say.